notebook/notebookapp.py

```diff
@@ -299,7 +299,8 @@ def init_handlers(self, settings):
         handlers.extend(load_handlers('services.kernelspecs.handlers'))
         handlers.extend(load_handlers('services.security.handlers'))
         handlers.extend(load_handlers('services.shutdown'))
-        
+        handlers.extend(settings['contents_manager'].get_extra_handlers())
```

Comment:
This is being added after the default files handler (the one which uses the contents manager). Doesn't that mean that this one (using static files) will never actually be used?

Reply:
I removed the global registration of the default files handler [here](https://github.com/jupyter/notebook/pull/2917/files#diff-aa5d3e97b16f9c0253d881fdec65c8c3R78), so this is where it gets registered, whichever class it is. On the base `ContentsManager`, it will be the generic `FilesHandler`, but on `FileContentsManager` subclasses it will be `AuthenticatedFileHandler`.

Reply:
The relevant change [below](https://github.com/jupyter/notebook/pull/2917/files#diff-c604f4531aba170341bbf6e9a155d5daR136) is that the base ContentsManager has `files_handler_class = FilesHandler` by default, so `get_extra_handlers` always returns something unless a ContentsManager explicitly overrides `files_handler_class = None`, in which case the `/files/` endpoint is disabled.

Reply:
Got it, thanks.